clients/src/main/java/org/apache/kafka/clients/NodeApiVersions.java

```diff
@@ -173,23 +186,52 @@ private String apiVersionToText(ApiVersion apiVersion) {
         }
 
         if (apiKey != null) {
-            Short usableVersion = usableVersions.get(apiKey);
-            if (usableVersion == NODE_TOO_OLD)
+            UsableVersion usableVersion = usableVersions.get(apiKey);
+            if (usableVersion.isTooOld())
                 bld.append(" [unusable: node too old]");
-            else if (usableVersion == NODE_TOO_NEW)
+            else if (usableVersion.isTooNew())
                 bld.append(" [unusable: node too new]");
             else
-                bld.append(" [usable: ").append(usableVersion).append("]");
+                bld.append(" [usable: ").append(usableVersion.defaultVersion).append("]");
         }
         return bld.toString();
     }
 
     public ApiVersion apiVersion(ApiKeys apiKey) {
-        for (ApiVersion nodeApiVersion : nodeApiVersions) {
-            if (nodeApiVersion.apiKey == apiKey.id) {
-                return nodeApiVersion;
-            }
+        for (UsableVersion usableVersion : usableVersions.values()) {
+            if (usableVersion.apiVersion.apiKey == apiKey.id)
+                return usableVersion.apiVersion;
```

Comment:
Can we just simply call `get` on the map? Also, maybe we should return `null` if there's no element (like `Map.get`).